### ISSUE TITLE
[TORCH] Support NLL loss with Poisson distribution of target

### ISSUE THREAD

**Issue by zahidwx** [llvm/torch-mlir#4209]
### Poisson NLL loss function
https://docs.pytorch.org/docs/stable/generated/torch.nn.PoissonNLLLoss.html

Currently this op is not supported in torch-mlir. It can be lowered to existing primitive ops in torch dialect.

**Comment by Prakhar-Dixit** [comment]
I am interested in taking up this issue.
Could you please assign it to me ?
@zahidwx

**Comment by Prakhar-Dixit** [comment]
@vivekkhandelwal1

**Comment by vivekkhandelwal1** [comment]
> ### Poisson NLL loss function
> https://docs.pytorch.org/docs/stable/generated/torch.nn.PoissonNLLLoss.html
> 
> Currently this op is not supported in torch-mlir. It can be lowered to existing primitive ops in torch dialect.

@zahidwx, I hope you are not working on this issue.

**Comment by zahidwx** [comment]
Hi @Prakhar-Dixit, I was planning to take this but you can go ahead and take this issue no problem.

**Comment by zahidwx** [comment]
PR: https://github.com/llvm/torch-mlir/pull/4232 merged to main.
